### ISSUE TITLE
Make default Torch profiler captures bounded and operation-focused

### ISSUE THREAD

**Issue by morluto** [morluto/flameox#367]
## Problem

The default `torch.profiler` capture profiles the entire Python entrypoint with CPU and CUDA activities while enabling shapes, allocation events, and Python stacks. Even a tiny bounded CUDA workload can therefore produce a hundreds-of-megabytes trace dominated by interpreter and framework startup rather than the operation being investigated.

Flameox reports that these features are expensive, but the zero-configuration path still enables all of them and offers no non-invasive warmup/active schedule. The only scheduled mode requires modifying the workload to import `flameox.sdk` and call `step()`.

## Reproduction

I captured two real Slime GAE workloads on an RTX 3060 with Torch 2.13.0 using the default adapter options:

```console
$ flameox capture run torch.profiler --workload slime-gae-sanitizer --workspace <workspace> --json
$ flameox capture run torch.profiler --workload slime-gae-sanitizer-target --workspace <workspace> --json
```

The first workload runs Slime's reference and chunked GAE implementations over two CUDA tensors shaped `[8, 512]`. The second runs only `chunked_gae`.

| Workload | CUDA kernels | CUDA kernel duration | Trace bytes | Capture wall time |
|---|---:|---:|---:|---:|
| reference + chunked | 2,618 | 3,306.7 µs | 528,778,647 | 16.26 s |
| chunked only | 57 | 104.6 µs | 518,398,685 | 12.65 s |

The chunked-only operation produced about 105 µs of kernel activity, yet its whole-entrypoint trace was 518 MB. Artifact registration succeeded, so this is not a failed or malformed producer case.

The plan confirms the defaults:

```json
{"mode":"whole_entrypoint","record_shapes":true,"profile_memory":true,"with_stack":true,"activities":["cpu","cuda_if_available"]}
```

Root cause in the current committed source:

- `src/flameox/adapters/torch_profiler.py` defaults `record_shapes`, `profile_memory`, and `with_stack` to true.
- `src/flameox/collectors/torch_launcher.py` wraps the complete target execution in one unscheduled profiler context.
- Scheduled profiling exists only in SDK mode, which requires workload source changes and explicit Flameox step calls.

PyTorch documents schedules with wait, warmup, and active periods and recommends warming past compilation and allocator initialization before profiling the section of interest:

- https://github.com/pytorch/pytorch/blob/main/docs/source/profiler.md
- https://github.com/pytorch/pytorch/blob/main/docs/source/user_guide/torch_compiler/torch.compiler_profiling_torch_compile.md

## Proposed outcome

Make the default capture bounded and useful before requiring SDK instrumentation. A suitable design could combine:

- conservative whole-entrypoint defaults, such as CUDA/operator timing without stacks, shapes, or allocation events unless requested;
- a preflight estimate or explicit confirmation when the selected feature set and scope have high cardinality;
- a non-invasive way to run warmup iterations outside the active profiler window when the workload exposes a repeatable command or region;
- region selection through standard PyTorch annotations such as `record_function`, or another workload-owned marker that does not require importing Flameox;
- artifact-size and event-count progress, with a clear stop/retry path using narrower options;
- generated follow-up options that add shapes, memory, or stacks only after a timing capture identifies a relevant operation.

SDK mode should remain available for precise training-loop schedules; the gap is a safe and useful first capture for existing workloads.

## Success criteria

- The default whole-entrypoint capture does not enable every high-cardinality feature.
- Planning states which enabled feature contributes which evidence and its expected cardinality/overhead.
- An unmodified workload can warm up and capture a bounded operation or annotated region.
- An agent can retry an oversized capture with a concrete narrower option set.
- The run records active duration/event counts and artifact size so trace overhead is auditable.
- Tests cover a small CUDA or CPU Torch workload where default trace growth remains proportional to the selected scope.

**Comment by morluto** [comment]
Consolidating the unique acceptance criteria from #346 here so the Torch-profiler admission design has one owner: include effective RSS, artifact-size, timeout, staging-growth, and output bounds in the plan; expose RSS/staging/trace-growth progress before hard termination; and preserve partial diagnostics plus process output when trace finalization exceeds a bound. Effective profiler options already belong to run semantics and should not gain a second schema.

**Comment by morluto** [comment]
Implemented on main in 73fde34. Whole-entrypoint Torch captures now default shapes, memory, stacks, FLOPs, and modules off; effective options and process, RSS, staging, output, timeout, and artifact bounds remain run semantics. Completed and partial captures preserve bounded provider diagnostics and process output, and recovery offers one lower-overhead replan. Explicit SDK mode owns wait, warmup, active, repeat, and step boundaries. Flameox deliberately does not infer an operation or warmup region from an unmodified workload.
